src/API/Google/Settings.php

```diff
@@ -387,6 +387,53 @@ public function get_store_address(): AccountAddress {
 		return $mc_address;
 	}
 
+	/**
+	 * Check whether the address has errors
+	 *
+	 * @param AccountAddress $address to be validated.
+	 *
+	 * @return array
+	 */
+	public function wc_address_errors( AccountAddress $address ): array {
+		/** @var WC $wc */
+		$wc = $this->container->get( WC::class );
+
+		$countries = $wc->get_wc_countries();
+
+		$locale          = $countries->get_country_locale();
+		$locale_settings = $locale[ $address->getCountry() ] ?? [];
+
+		$fields_to_validate = [
+			'address_1' => $address->getStreetAddress(),
+			'city'      => $address->getLocality(),
+			'country'   => $address->getCountry(),
+			'postcode'  => $address->getPostalCode(),
+			'state'     => $address->getRegion(),
+		];
+
+		return $this->validate_address( $fields_to_validate, $locale_settings );
+	}
+
+	/**
+	 * Check whether the required address fields are empty
+	 *
+	 * @param array $address_fields to be validated.
+	 * @param array $locale_settings locale settings
+	 * @return array
+	 */
+	public function validate_address( array $address_fields, array $locale_settings ): array {
+		$errors = array_filter(
+			$address_fields,
+			function ( $field ) use ( $locale_settings, $address_fields ) {
+				$is_required = $locale_settings[ $field ]['required'] ?? true;
```

Comment:
If the default is `false`, the postcode would not be required for countries like [UK](https://github.com/woocommerce/woocommerce/blob/4cf10ad2174de8bde62fa1b753691e2f091db413/plugins/woocommerce/includes/class-wc-countries.php#L1416), [ES](https://github.com/woocommerce/woocommerce/blob/4cf10ad2174de8bde62fa1b753691e2f091db413/plugins/woocommerce/includes/class-wc-countries.php#L1343), [FR](https://github.com/woocommerce/woocommerce/blob/4cf10ad2174de8bde62fa1b753691e2f091db413/plugins/woocommerce/includes/class-wc-countries.php#L996) etc and that would be incorrect.

Singapore seems to use postcodes, see: https://en.wikipedia.org/wiki/Postal_codes_in_Singapore

[In the original implementation](https://github.com/woocommerce/google-listings-and-ads/blob/48df4fb68654be8c6952149aba9cfd0623ef7fba/js/src/hooks/useStoreAddress.js#L69), the postcode was always required in contrast using the class WC_Countries we are able to allow countries like Ireland, United Arab Emirates, Uganda etc be able to save the WC store without using the postcode.

Anyway, I think it is a good idea to show the errors as was mentioned in #634 so the users know the reason why the button is disabled.

Reply:
> 3. Edit your store address in WC Settings, choose one country where is not required postcode such as Ireland, Singapur, Emirates and leave blank the postcode.

I followed the test instructions to set Singapore as WC store address, and then I could not finish the onboarding setup. So I thought it was a problem. 😅 

> Singapore seems to use postcodes, see: https://en.wikipedia.org/wiki/Postal_codes_in_Singapore

Seems like the `required` postcode of locale settings in WC core is not quite correct to be treated as a must-required field. In Taiwan, we do have postcodes but it's not required address info for most cases, a postcode is still recommended to be filled when providing address though.

I think it's probably a trade-off when choosing to make it required or not when a country is not really required a postcode to identify an address. Just mainly depends on the services to be provided, which way is preferred.  

Consider the WC Settings page would have the postcode input field when the `required` is `true` or `NULL`, so it would not totally block users to finish the onboarding setup. It should be ok to keep it as is.

Reply:
Using WC_Countries allow us to be more flexible with the postcode requirement, with the previous implementation the postcode was always required and even countries that have the postcode requirement = `false` in WC could not complete the onboarding process.

Reply:
Yes. So it should be ok to keep using `true` as default. This thread should be no longer a concern. Thanks for the finding and elaborations! 😄